openshift/hni-patroni-postgresql-deploy.yml

```diff
@@ -0,0 +1,263 @@
+                timeoutSeconds: 5
+                failureThreshold: 4
+                exec:
+                  command:
+                    - /usr/share/scripts/patroni/health_check.sh
+              volumeMounts:
+                - mountPath: /home/postgres/pgdata
+                  name: postgresql
+          dnsPolicy: ClusterFirst
+          restartPolicy: Always
+          schedulerName: default-scheduler
+          securityContext: {}
+          serviceAccountName: ${NAME}
+          terminationGracePeriodSeconds: 0
+      updateStrategy:
+        type: RollingUpdate
+      volumeClaimTemplates:
+        - metadata:
+            annotations:
+              volume.beta.kubernetes.io/storage-class: ${STORAGE_CLASS}
+            labels:
+              app: ${APP_LABEL_NAME}
+            name: postgresql
+          spec:
+            storageClassName: ${STORAGE_CLASS}
+            accessModes:
+              - ReadWriteOnce
+            resources:
+              requests:
+                storage: ${PVC_SIZE}
+parameters:
+  - name: NAME
+    description: |
+      The name of the application for labelling all artifacts.
+    displayName: Application Name
+    value: hni-patroni-postgresql
+  - name: REPLICAS
+    description: |
+      The number of StatefulSet replicas to use.
+    displayName: REPLICAS
+    value: "3"
+  - name: CPU_REQUEST
+    description: |
+      Starting amount of CPU the container can use.
+    displayName: CPU REQUEST
+    value: "250m"
+  - name: CPU_LIMIT
+    description: Maximum amount of CPU the container can use.
+    displayName: CPU Limit
+    value: "1"
+  - name: MEMORY_REQUEST
+    description: Starting amount of memory the container can use.
+    displayName: Memory Request
+    value: 512Mi
+  - name: MEMORY_LIMIT
+    description: Maximum amount of memory the container can use.
+    displayName: Memory Limit
+    value: 512Mi
+  - name: IMAGE_STREAM_NAMESPACE
+    description: |
+      The OpenShift Namespace where the patroni and postgresql
+      ImageStream resides.
+    displayName: ImageStream Namespace
+    value: bcgov
+  - name: IMAGE_NAME
+    description: |
+      The Patroni image stream name
+    value: patroni-postgres
+  - name: IMAGE_TAG
+    description: |
+      The image tag used to specify which image you would like deployed.
+      Don't use `latest`.
+    value: "12.4-latest"
+  - name: PVC_SIZE
+    description: The size of the persistent volume to create.
+    displayName: Persistent Volume Size
+    value: 1Gi
+  - name: STORAGE_CLASS
+    value: netapp-file-standard
+  - name: IMAGE_REGISTRY
+    value: image-registry.openshift-image-registry.svc:5000
+  - displayName: App label name
+    description: |
+      The App label name used for all components of the application. This is useful
+      in dev and test when you want to clear all and have a clean working namespace.
+    name: APP_LABEL_NAME
+    required: true
+    value: hni-patroni-postgresql
```

Comment:
We assign application name as label to all the resources for that app. For example app: hnsesb-dev. 
May be this will be handy when we integrate patroni setup with hnsesb setup that we should assign same label to resources created for Patroni